packages/core/src/dialects/abstract/query-generator-typescript.ts

```diff
@@ -1080,4 +1084,113 @@ Only named replacements (:name) are allowed in literal() because we cannot guara
   }
+
+  /**
+   * Returns a query that creates a savepoint.
+   *
+   * @param savepointName
+   */
+  createSavepointQuery(savepointName: string): string {
+    if (!this.dialect.supports.savepoints) {
+      throw new Error(`Savepoints are not supported by ${this.dialect.name}.`);
+    }
+
+    return `SAVEPOINT ${this.quoteIdentifier(savepointName)}`;
+  }
+
+  /**
+   * Returns a query that starts a transaction.
+   *
+   * @param options
+   */
+  startTransactionQuery(options?: StartTransactionQueryOptions): string {
+    if (this.dialect.supports.connectionTransactionMethods) {
+      throw new Error(`startTransactionQuery is not supported by the ${this.dialect.name} dialect.`);
+    }
+
+    if (options) {
+      const START_TRANSACTION_QUERY_SUPPORTED_OPTIONS = new Set<keyof StartTransactionQueryOptions>();
+
+      if (this.dialect.supports.startTransaction.readOnly) {
+        START_TRANSACTION_QUERY_SUPPORTED_OPTIONS.add('readOnly');
+      }
+
+      if (this.dialect.supports.startTransaction.transactionType) {
+        START_TRANSACTION_QUERY_SUPPORTED_OPTIONS.add('transactionType');
+      }
+
+      rejectInvalidOptions(
+        'startTransactionQuery',
+        this.dialect.name,
+        START_TRANSACTION_QUERY_SUPPORTABLE_OPTIONS,
+        START_TRANSACTION_QUERY_SUPPORTED_OPTIONS,
+        options,
+      );
+    }
+
+    return joinSQLFragments([
+      this.dialect.supports.startTransaction.useBegin ? 'BEGIN' : 'START',
+      'TRANSACTION',
+      options?.readOnly ? 'READ ONLY' : '',
+    ]);
+  }
+
+  /**
+   * Returns a query that commits a transaction.
+   */
+  commitTransactionQuery(): string {
+    if (this.dialect.supports.connectionTransactionMethods) {
+      throw new Error(`commitTransactionQuery is not supported by the ${this.dialect.name} dialect.`);
+    }
+
+    return 'COMMIT';
+  }
+
+  /**
+   * Returns a query that rollbacks a savepoint.
+   *
+   * @param savepointName
+   */
+  rollbackSavepointQuery(savepointName: string): string {
+    if (!this.dialect.supports.savepoints) {
+      throw new Error(`Savepoints are not supported by ${this.dialect.name}.`);
+    }
+
+    return `ROLLBACK TO SAVEPOINT ${this.quoteIdentifier(savepointName)}`;
+  }
+
+  /**
+   * Returns a query that rollbacks a transaction.
+   */
+  rollbackTransactionQuery(): string {
+    if (this.dialect.supports.connectionTransactionMethods) {
+      throw new Error(`rollbackTransactionQuery is not supported by the ${this.dialect.name} dialect.`);
+    }
+
+    return 'ROLLBACK';
+  }
+
+  /**
+   * Returns a query that sets the transaction isolation level.
+   *
+   * @param isolationLevel
+   */
+  setIsolationLevelQuery(isolationLevel: IsolationLevel): string {
+    if (!this.dialect.supports.isolationLevels) {
+      throw new Error(`Isolation levels are not supported by ${this.dialect.name}.`);
+    }
+
+    if (this.dialect.supports.connectionTransactionMethods) {
```

Comment:
this condition should be inverted

Reply:
Done

Reply:
Ooooh wait, `connectionTransactionMethods` means the dialect uses JS methods on the connector, I read it as its exact opposite. Please ignore my previous comment